snuba/subscriptions/codecs.py

```diff
@@ -62,3 +69,44 @@ def encode(self, value: SubscriptionTaskResult) -> KafkaPayload:
             ).encode("utf-8"),
             [],
         )
+
+
+class SubscriptionScheduledTaskEncoder(Encoder[bytes, ScheduledTask[Subscription]]):
+    """
+    Encodes/decodes a scheduled subscription to bytes for Kafka.
+    Does not support non SnQL subscriptions.
+    """
+
+    def __init__(self, entity_key: EntityKey) -> None:
+        self.__entity_key = entity_key
+
+    def encode(self, value: ScheduledTask[Subscription]) -> bytes:
+        assert isinstance(value.task.data, SnQLSubscriptionData)
+
+        return cast(
+            bytes,
+            rapidjson.dumps(
+                {
+                    "timestamp": value.timestamp.isoformat(),
```

Comment:
Why do we need to format the date every time? Can't we pass the timestamp as integer instead ? The date format will have an impact when you will try to schedule tons of these at the same time.

Reply:
`value.timestamp` is not an integer to begin with, it's a Python datetime object so it would to be formatted each time we schedule. We could format the date as an integer but that wouldn't be any faster than this. The other subscription codec uses isoformat() for timestamp so I think it's better to keep it consistent.

Reply:
Considering the internal representation (https://github.com/python/cpython/blob/v3.7.0/Include/datetime.h#L10-L22) , I'd bet it will be more efficient to encode and decode into timestamp than into formatted string. 
Though I guess it would have a much bigger impact on decoding (parsing the string to the internal representation), that part is less critical as it would be done on the executor, so maybe we can get away on this. 

Still we should revisit this when we profile the scheduler and we see the actual impact.